src/AssembleElemSolverAlgorithm.C

```diff
@@ -112,25 +110,23 @@ AssembleElemSolverAlgorithm::execute()
       for(int simdElemIndex=0; simdElemIndex<smdata.numSimdElems; ++simdElemIndex) {
         extract_vector_lane(smdata.simdrhs, simdElemIndex, smdata.rhs);
         extract_vector_lane(smdata.simdlhs, simdElemIndex, smdata.lhs);
-        for (int ir=0; ir < rhsSize_; ++ir)
-          smdata.lhs(ir, ir) /= diagRelaxFactor_;
-        apply_coeff(nodesPerEntity_, smdata.ngpElemNodes[simdElemIndex],
+        for (int ir=0; ir < rhsSize; ++ir)
+          smdata.lhs(ir, ir) /= diagRelaxFactor;
+        (*deviceCoeffApplier)(nodesPerEntity, smdata.ngpElemNodes[simdElemIndex],
                     smdata.scratchIds, smdata.sortPermutation, smdata.rhs, smdata.lhs, __FILE__);
       }
 #else
```

Comment:
Is there a reason to keep this `#ifdef`? Or can we collapse this?

Reply:
That ifdef is around the simd-loop, we probably could collapse that. Possibly set num-simd-elems to 1 if cuda or something like that?